Doc/library/contextlib.rst

```diff
@@ -314,7 +314,8 @@ Functions and classes provided:
 
    If the code within the :keyword:`!with` block raises a
    :exc:`BaseExceptionGroup`, suppressed exceptions are removed from the
-   group.  If any exceptions in the group are not suppressed, a group containing them is re-raised.
+   group.  If any exceptions in the group are not suppressed, a group containing
+   them is re-raised (see also :meth:`~BaseExceptionGroup.derive`).
```

Comment:
```suggestion
   them is re-raised using the group's :meth:`~BaseExceptionGroup.derive` method.
```

Reply:
That wording is not quite correct - derive is not what re-raises it, it plays a part in the process.

Reply:
Also, it may not use the BaseExceptionGroup.derive method, if you overwrite it in the subclass (the point here is that you should consider overriding it).

Reply:
> Also, it may not use the BaseExceptionGroup.derive method, if you overwrite it in the subclass

I understood that, but it will render as `derive()` in the docs rather than `BaseException.derive()` -- the `~` ensures that.

Maybe something like this?

```suggestion
   them is created using the old group's :meth:`~BaseExceptionGroup.derive`
   method, and the new group is then re-raised.
```